### DISCUSSION TITLE
Getting Error When using Celery Worker

### DISCUSSION THREAD

**Discussion by Asturias-sam** [apache/superset#14145] [Q&A / Help]
I have configured the celery worker to run the queries in async mode, but when i am running the queries in sql lab they are getting stuck and getting the below stack trace from logs:

```
Traceback (most recent call last):
  File "/opt/superset/venv/lib/python3.7/site-packages/superset/result_set.py", line 144, in __init__
    series, type=pa.timestamp("ns", tz=tz)
  File "pyarrow/array.pxi", line 859, in pyarrow.lib.Array.from_pandas
  File "pyarrow/array.pxi", line 265, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 80, in pyarrow.lib._ndarray_to_array
  File "pyarrow/error.pxi", line 107, in pyarrow.lib.check_status
pyarrow.lib.ArrowTypeError: Did not pass numpy.dtype object
ERROR:superset.result_set:Did not pass numpy.dtype object
```

Can anyone help me with that ?

**Comment by junlincc** [comment]
@robdiciuccio Rob, maybe you can help?

**Comment by robdiciuccio** [comment]
@Asturias-sam There's a `try/except` wrapper around [this code](https://github.com/preset-io/superset/blob/master/superset/result_set.py#L146:L146), so it should only be logging the exception and continuing to process the query. Are you able to run the query directly against the database? Is it a long-running query?

**Reply by Asturias-sam** [reply]
Hello Rob,

When i disable the async mode i able to run the queries but once i enable the query in async i am seeing queries are getting stuck.
No they are not long running queries usually they take 1-2 minute.
